File: DataAccessLayer/Entities/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataAccessLayer.Entities
{
    public class Product : BaseEntity
    {
        [Required, StringLength(100)]
        public string Title { get; set; }

        [StringLength(200)]
        public string Description { get; set; }

        [Required]
        public ProductType Type { get; set; }

        [Required]
        public int WaterPointId { get; set; }

        [StringLength(50)]
        public string photoUrl { get; set; }

        [Required]
        public decimal Price { get; set; }

        [Required]
        public decimal QuantityAvailable { get; set; }

        public WaterPoint WaterPoint { get; set; }  

        public List<OrderDetail> OrderDetails { get; set; }
    }


    public enum ProductType
    {
        StillWater,
        SparklingWater,
        IndustrialWater,
        Other
    }
}

File: UI-Angular/src/models/product.ts


export class Product{
    id!: number;
    title: string;
    description: string;
    type: ProductType;
    waterPointId: number;
    price: number;
    quantityAvailable: number;
    photoUrl: string;

    constructor(title: string, description: string, type: ProductType, 
    waterPointId: number, price: number, quantityAvailable: number, photoUrl: string){
        this.title = title;
        this.description = description;
        this.type = type;
        this.waterPointId = waterPointId;
        this.price = price;
        this.quantityAvailable = quantityAvailable;
        this.photoUrl = photoUrl;
    }
}

export enum ProductType{
        StillWater,
        SparklingWater,
        IndustrialWater,
        Other
}